tests/Doctrine/Tests/DBAL/Types/DateImmutableTypeTest.php

```diff
@@ -9,13 +9,13 @@
 use Doctrine\DBAL\Types\ConversionException;
 use Doctrine\DBAL\Types\DateImmutableType;
 use Doctrine\DBAL\Types\Type;
+use PHPUnit\Framework\MockObject\MockObject;
 use PHPUnit\Framework\TestCase;
-use Prophecy\Prophecy\ObjectProphecy;
 use function get_class;
 
 class DateImmutableTypeTest extends TestCase
 {
-    /** @var AbstractPlatform|ObjectProphecy */
+    /** @var AbstractPlatform|MockObject */
```

Comment:
Not important, but we should start using `&MockObject`

Reply:
Is it the time for PHPStan to drop the conversion of `|` to `&` in tests? At least if they had made it possible to disable this behavior, we could disable it and make sure all our types are declared correctly.

Reply:
Hmm, not sure how the PHPStan PHPUnit plugin works here: `&MockObject` is the correct type here, but let's leave it for another patch/discussion.